include/eve/detail/skeleton.hpp

```diff
@@ -54,7 +54,11 @@ namespace eve::detail
 
   // Compute a transformed wide type
   template<typename F, typename... Ts>
-  struct wide_result
+  struct wide_result;
+
+  template<typename F, typename... Ts>
+  requires requires { std::declval<F>()(at(std::declval<Ts>(), 0)...); }
```

Comment:
We need to sfinae the case the scalar type deduction is non sensicak. This requires is just a way to easily prune out the wrong cases.

Reply:
I think i need more explaining here. let's talk on slack

Reply:
From Slack:

 Map has ot be called on emulated types, think `wide<double>` on ARM32 for ex. The principles of `map(f,x...)` is to compute the result as being a wide of the proper cardinal and type which is computed as the SCALAR return type of f.  So we do compute `decltype( f( <every element_type of all x>() ... )`

However, as the EVE computing model is that function on scalar only exists if all their parameters are the same, you may end up with scenario where calling map would be an error.

Case in point :  let's say you're on ARM32 and you want to call `eve::max( 1, some_wide_double );`
As `wide<double>` is emulated on ARM32, we go through map. And the map code tries to compute `decltype(max(int,double));` which is not a thing.

So this type mistake is captured by the require, it ADL out map and we go through the regular function call. Guess what, the regular call see int  + wide<double> and has code to call itself with max(wide<double>, wide<double>)
This will also trigger a map deduction that this time will work and do the correct thing.